spyderlib/plugins/configdialog.py

```diff
@@ -903,77 +917,419 @@ def _save_lang(self):
+        manage_layout.addLayout(buttons_preview_layout)
+        manage_group = QGroupBox(_("Manage color schemes"))
+        manage_group.setLayout(manage_layout)
+
+        vlayout.addWidget(manage_group)
         self.setLayout(vlayout)
-        
-    @Slot(str)
-    def reset_to_default(self, name):
-        set_default_color_scheme(name, replace=True)
-        self.load_from_conf()
-            
+
+        # Signals and slots
+        create_button.clicked.connect(self.create_new_scheme)
+        edit_button.clicked.connect(self.edit_scheme)
+        self.reset_button.clicked.connect(self.reset_to_default)
+        self.delete_button.clicked.connect(self.delete_scheme)
+        self.schemes_combo.currentIndexChanged.connect(self.update_preview)
+        self.schemes_combo.currentIndexChanged.connect(self.update_buttons)
+
+        # Setup
+        for name in names + custom_names:
+            self.scheme_editor_dialog.add_color_scheme_stack(name)
+
+        self.update_combobox()
+
     def apply_settings(self, options):
         self.main.editor.apply_plugin_settings(['color_scheme_name'])
         if self.main.historylog is not None:
             self.main.historylog.apply_plugin_settings(['color_scheme_name'])
         if self.main.inspector is not None:
             self.main.inspector.apply_plugin_settings(['color_scheme_name'])
+        self.update_preview()
+        self.update_combobox()
+
+    # Helpers
+    # -------------------------------------------------------------------------
+    @property
+    def current_scheme(self):
+        return self.schemes_combo.currentText().lower()
+
+    @property
+    def current_scheme_index(self):
+        return self.schemes_combo.currentIndex()
+
+    def update_combobox(self):
+        """Recreates the combobox contents."""
+        index = self.current_scheme_index
+        self.schemes_combo.blockSignals(True)
+        names = self.get_option("names")
+        custom_names = self.get_option("custom_names")  # 'custom-n'
+
+        if custom_names:
+            choices = names + [None] + sorted(custom_names)
+            scheme_choices = list(zip(choices, [a for a in choices]))
+        else:
+            choices = sorted(names)
+            scheme_choices = list(zip(choices, [a.lower() for a in choices]))
+
+        combo = self.schemes_combo
+        combo.clear()
+
+        for item in scheme_choices:
+            name, key = item
+            if name is None and key is None:
+                continue
+            combo.addItem(name, key)
+
+        if custom_names:
+            combo.insertSeparator(len(names))
+
+        self.schemes_combo.blockSignals(False)
+        self.schemes_combo.setCurrentIndex(index)
+
+    def update_buttons(self):
+        """Updates the enable status of delete and reset buttons."""
+        current_scheme = self.current_scheme.lower()
+        names = [n.lower() for n in self.get_option("names")]
+        delete_visible = current_scheme not in names

```

Comment:
This variable should be named `delete_enabled`, not `delete_visible` because we're not hiding the Delete button, but enabling or disabling it.